### PR TITLE
RENDERER: Add FXAA option.

### PR DESCRIPTION
**Pull request by dsvensson**
Default disabled, ideally used with vid_framebuffer 2 to avoid adding AA to HUD.

NVIDIA FXAA Shader v3.11, BSD license, imported verbatim.

### CODE REVIEW

**Comment by dsvensson** [comment]
Some samples:

image
image
image
image

File: help_variables.json [hunk 1]
```diff
@@ -21781,6 +21781,86 @@
       "remarks": "Setting vid_framebuffer_scale overrides custom FBO width/height.",
       "type": "integer"
     },
+    "vid_framebuffer_fxaa": {
+      "default": "0",
+      "desc": "Fast Approximate Anti-Aliasing. Enable vid_framebuffer 2 to avoid filtering HUD. Typically between 1-6.",
+      "group-id": "52",
+      "type": "boolean",
```
Comment (phylter-qw) [inline help_variables.json:21788]: not an integer?

File: src/glsl/post_process_screen.fragment.glsl [hunk 2]
```diff
@@ -2,6 +2,2079 @@
...
+------------------------------------------------------------------------------
+FXAA3 requires RGBL as input unless the following is set,
+
+  #define FXAA_GREEN_AS_LUMA 1
+
+In which case the engine uses green in place of luma,
+and requires RGB input is in a non-linear colorspace.
+
+RGB should be LDR (low dynamic range).
+Specifically do FXAA after tonemapping.
```
Comment (phylter-qw) [inline src/glsl/post_process_screen.fragment.glsl:136]: He says here it's important to apply FXAA after tone mapping and gamma correction, so the input is in a non-linear color space.

In the shader main() function down below, FXAA is being applied _before_ tone mapping.

I'm not sure where gamma correction is applied. Maybe the color buffer is sRGB so it happens automatically. Does anyone know?

**Review by phylter-qw — COMMENTED** [review]
Much thanks. Such impress. Couple minor scruples I pointed out.

File: help_variables.json (hunk 1 above)
Comment (dsvensson) [reply help_variables.json:21788]: thanks 😂

File: src/glsl/post_process_screen.fragment.glsl (hunk 2 above)
Comment (dsvensson) [reply src/glsl/post_process_screen.fragment.glsl:136]: Yep, that's correct. But it's somewhat annoying to do, and I couldn't spot any difference. The gamma can be applied via `vid_software_palette 1` at the end of the shader, but if you're using hw gamma you're not going to be doing it afterwards anyways.

Comment (dsvensson) [reply src/glsl/post_process_screen.fragment.glsl:136]: Actually tonemap is only applied to world at least, so Fxaa could move to after that. Could also play with applying software palette before fxaa.

Comment (dsvensson) [reply src/glsl/post_process_screen.fragment.glsl:136]: But FxaaPixelShader takes sampler2D arg, so I guess the intended situation is to have a pipeline where the world is rendered to a texture tonemapped, and then a fxaa shader is invoked.

Comment (dsvensson) [reply src/glsl/post_process_screen.fragment.glsl:136]: I guess I should add the reasoning here in a comment to be one step ahead of future confusion when the next person read the code.

**Pull request merged**